test/test-mpl2-license-header.js

```diff
@@ -0,0 +1,89 @@
+// Note: This line is here intentionally, to break MPL2_LICENSE_TEST
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+"use strict";
+
+const { Cc, Ci, Cu } = require("chrome");
+const options = require('@loader/options');
+const { id } = require("sdk/self");
+const { getAddonByID } = require("sdk/addon/manager");
+const { mapcat, map, filter, fromEnumerator } = require("sdk/util/sequence");
+const { readURISync } = require('sdk/net/url');
+
+const ios = Cc['@mozilla.org/network/io-service;1'].
+              getService(Ci.nsIIOService);
+
+ const MIT_LICENSE_HEADER = [];
+
+const MPL2_LICENSE_TEST = new RegExp([
+  "^\\/\\* This Source Code Form is subject to the terms of the Mozilla Public",
+  " \\* License, v\\. 2\\.0\\. If a copy of the MPL was not distributed with this",
+  " \\* file, You can obtain one at http:\\/\\/mozilla\\.org\\/MPL\\/2\\.0\\/\\. \\*\\/"
+].join("\n"));
+
+// Note: Using regular expressions because the paths a different for cfx vs jpm
+const IGNORES = [
+  /lib\/(diffpatcher|method)\/.+$/, // MIT
+  /lib\/sdk\/fs\/path\.js$/, // MIT
+  /lib\/sdk\/system\/child_process\/.*/,
+  /tests?\/buffers\/.+$/, // MIT
+  /tests?\/path\/test-path\.js$/,
+  /tests?\/querystring\/test-querystring\.js$/,
+];
+
+const ignoreFile = file => !!IGNORES.find(regex => regex.test(file));
+
+const baseURI = "resource://test-sdk-addon/";
+
+const uri = (path="") => baseURI + path;
+
+const toFile = x => x.QueryInterface(Ci.nsIFile);
+const isTestFile = ({ path, leafName }) => {
+  return !ignoreFile(path) && /\.jsm?$/.test(leafName)
+};
+const getFileURI = x => ios.newFileURI(x).spec;
+
+const getDirectoryEntries = file => map(toFile, fromEnumerator(_ => file.directoryEntries));
```

Comment:
Seems like it might be useful to have fromEnumerator take an iid to QI everything to